### PR TITLE
Use LYS API endpoint for woocommerce_admin_launch_your_store_survey_completed option

### LINKED ISSUE #47800: [LYS] `get_options` warning notice
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

We're using get_options API to fetch the `woocommerce_admin_launch_your_store_survey_completed` option value but this option is not added to allow list in the `get_options` function. So, we're getting the warning notice.

I'm not sure why we missed this option in the allow list because the API should return an error in a development environment. :thinking:

```
The Automattic\WooCommerce\Admin\API\Options::get_options function is deprecated since version 6.3.
```

However, since the get_options API has been deprecated, we should better create a dedicated API to fetch the option value.

### Expected behavior

No warning notice should be displayed when fetching the `woocommerce_admin_launch_your_store_survey_completed` option value.

### Actual behavior

Warning notice is displayed when fetching the `woocommerce_admin_launch_your_store_survey_completed` option value.

### Steps to reproduce

1. Use WooCommerce trunk or 9.0.0 beta release
2. Go to WooCommerce > Home > Launch your store task
3. Complete the task
4. Confirm the warning notice is displayed in the debug.log file in production environment or 403 Forbidden in development environment

### WordPress Environment

n/a

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #47800 

This PR replaces the use of options endpoint with the LYS API endpoint to query woocommerce_admin_launch_your_store_survey_completed option, as the options endpoint is deprecated.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch. 
2. SSH into the JN site.
3. cd into `wp-content` and run `tail -f debug.log`
4. Go to `WooCommerce -> Home`
5. Click on `Launch your store` task
6. Click on `Launch your store` button.
7. Confirm there is no warning notice from the log.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Replace the use of options endpoint with the LYS API endpoint to query woocommerce_admin_launch_your_store_survey_completed option.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/transitional/services.tsx [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
  */
-import { OPTIONS_STORE_NAME } from '@woocommerce/data';
-import { resolveSelect } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
 
 export const fetchSurveyCompletedOption = async () =>
-	resolveSelect( OPTIONS_STORE_NAME ).getOption(
-		'woocommerce_admin_customize_store_survey_completed'
-	);
+	await apiFetch( {
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/customize-store/transitional/services.tsx]: I think this change to CYS is unintended?

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Thanks for the fix, @moon0326! I believe the changes made to CYS is unintentional.

Also, it seems like there are multiple requests fired, perhaps we can check if it's already fetched?

image

File: plugins/woocommerce-admin/client/customize-store/transitional/services.tsx (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/customize-store/transitional/services.tsx]: Thank you for the catch @ilyasfoo 

My bad 🤦 I'll update it soon

Comment (moon0326) [reply plugins/woocommerce-admin/client/customize-store/transitional/services.tsx]: @ilyasfoo Updated in [Revert back unwanted changes](https://github.com/woocommerce/woocommerce/pull/47915/commits/f8477113c292d1c65c737870c53eac01c18e3471)

[Cache /wc-admin/launch-your-store/survey-completed call](https://github.com/woocommerce/woocommerce/pull/47915/commits/b63e7e713371020d9a69e7a78a2d5339ad6f4dcd) also implements a cache for the call.

**Review by ilyasfoo — APPROVED** [review]
Thanks, @moon0326! 

Nitpicky: It seems the cache is not 100% perfect, if say the apiFetch is in progress, it will still fetch twice (reproduced it by refreshing at the congrats page). Still, I don't think it's a blocker.

image

**Comment by moon0326** [comment]
> Thanks, @moon0326!
> 
> Nitpicky: It seems the cache is not 100% perfect, if say the apiFetch is in progress, it will still fetch twice (reproduced it by refreshing at the congrats page). Still, I don't think it's a blocker.
> 
> image

Good catch! We might want to convert it to a store to fix it 🤔 I'm going to merge it for now.

**Pull request merged**